src/main/java/org/jenkinsci/remoting/util/AnonymousClassWarnings.java

```diff
@@ -63,7 +69,7 @@ public static void check(@NonNull Class<?> clazz) {
         } else {
             // May not call methods like Class#isAnonymousClass synchronously, since these can in turn trigger remote class loading.
             try {
-                channel.executor.submit(() -> doCheck(clazz));
+                threadPool.submit(() -> doCheck(clazz));
```

Comment:
as we are not sure if this is the root cause or not - would it make sense to control which threadpool to use based on a system propery (so we can either try the new code, or revert to the old behaviour depending on the default?)